.github/workflows/ci.yml

```diff
@@ -99,14 +99,29 @@ jobs:
     - name: Set up pandas
       uses: ./.github/actions/setup
 
+    # Github workflows built-in path filters
+    # don't work on a level of individual jobs or steps.
+    # So, we use this.
+    - uses: dorny/paths-filter@v2
```

Comment:
I really did not want to add an external "action", but as the comment says, github action don't work on the individual step level, only at the whole workflow.

So, either that this stays like that. Or we will have a separate workflow file for this.
I don't mind doing it either way, just LMK what is preferred :)